java-checks/src/test/files/checks/HardCodedCredentialsCheck.java

```diff
@@ -30,15 +32,25 @@ private void a() {
     myA.setProperty("xxxxx", "password");
     myA.setProperty(12, "xxxxx");
     myA.setProperty(new Object(), new Object());
-    
+
     MyUnknownClass.myUnknownMethod("password", "xxxxx"); // Noncompliant
+
+    PasswordAuthentication pa;
+    pa = new PasswordAuthentication("userName", "1234".toCharArray());  // Noncompliant
+    pa = new PasswordAuthentication("userName", pwd); // Compliant
+    pa = new PasswordAuthentication("userName", getPwd(var)); // Compliant
+    pa = new PasswordAuthentication("userName", var.toCharArray()); // Compliant
+  }
```

Comment:
`pa = new SomeOtherAuthentication("username", "1234".toCharArray());` could also be tested to show that there are three conditions that needs to be satisfied.